<p><img src="https://files.cdn.thinkific.com/file_uploads/133292/images/a9e/c1a/bc3/P02.jpg"></p>

`MCSDSC02P01V3______`

`MCSDSC02P01V2______`

`MCSDSELIMC02P01V1010621`

<h3><font color="#FF0F00">Комментарий ассистента</font></h3>

Привет! Учитывая, что сейчас на курсе нет теории по предобратке на Pandas (не считая модуля в курсе по ML), твоя итерация очень достойная!

Как обычно, заголовки (комментариев) будут трех цветов. <font color="89DD00">Зеленый</font> — все правильно (и если комментария нет, то это также значит, что все правильно). <font color="FFB800">Желтый</font> — можно доработать. <font color="FF0F00">Красный</font> — нужно исправить.

Постарайся учесть эти комментарии в ходе доработки проекта и не удаляй их.

## Постановка задачи

Предоставлена информация об использовании приложении предварительного заказа столиков в ректоранах на территории дизайн-завода Флакон и Хлебозавода №9 с возможностью предзаказа блюд к приходу. Необходимо изучить данные сервиса с тестового запускаи ответить на вопросы, поставленные менеджером проекта:

- На какие кафе приходится больше всего заказов? А на какие дни?
- Чаще ли приложением пользуются именно в обеденное время?
- Насколько большими компаниями  (или наоборот поодиночке) приходят в кафе?
### План работы:
 1. Подготовить датасет: 
        1. Привести к общим значениям и формату наименования столбцов и кафе.
        2. Удалить дубликаты.
        3. Проверить на наличие  пропусков.
 2. Рассмотреть распределение заказов: 
    1. Посчитать общее количество заказов для каждого ресторана, выявить популярные.
    2. Выявить распределение бронирований по дням недели.
    3. Проанализировать время посещения.
    4. Найти наиболее частое количество персон за заказанным столиком.
 3. Сформулировать выводы на основании полученных данных.

<h3><font color="#FFB800">Комментарий ассистента</font></h3>

Желательно еще добавить строчку про сам сервис — что он из себя представляет, что делает.

В остальном тут ОК.

## Импорт данных и библиотек

In [1]:
import pandas as pd
from datetime import datetime as dt

Прочитаем файл `bez_ocheredi.csv` и сохраним его в переменную `bo_data`.

In [2]:
bo_data=pd.read_csv("../../data/bez_ocheredi.csv")

<h3><font color="#FFB800">Комментарий ассистента</font></h3>

Название `df` не позволяет понять какие именно данные в нем записаны. Желательно учесть рекомендации и назвать датафрейм, например, `bo_data`.

Получим первые 10 строк таблицы.

In [3]:
bo_data.head(10)

userID    ресторан     weekday  ordered    Столик для 
0  b912c3c6a6ad   Iskra Bar     Monday     10:00             1
1  2dac0150717d   Iskra Bar     Monday     13:30             2
2  ba83aefab5dc   Iskra Bar    Tuesday     12:30             3
3  02996f1dd2ea   Iskra Bar  Wednesday     12:00             2
4  7c90be56d2ab   Iskra Bar   Thursday     14:30             3
5  51e0962387f7      Там Ем     Friday     12:30             2
6  c4f260a2b48a         NaN   Saturday     15:00             6
7  6b71309d6a8a  Крафтейнер     Monday     10:30             1
8  c5b969cb63a2  Бар Сибирь  Wednesday     15:30             2
9  b1570962e68c  Бар Сибирь  Wednesday     12:30             3

Получим общую информацию о данных из таблицы.

In [4]:
bo_data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1024 entries, 0 to 1023
Data columns (total 5 columns):
 #   Column        Non-Null Count  Dtype 
---  ------        --------------  ----- 
 0   userID        1024 non-null   object
 1    ресторан     959 non-null    object
 2    weekday      1024 non-null   object
 3    ordered      1024 non-null   object
 4    Столик для   1024 non-null   int64 
dtypes: int64(1), object(4)
memory usage: 40.1+ KB


<h3><font color="#FFB800">Комментарий ассистента</font></h3>

Самым оптимальным для этого набора данных будет метод `info()`. Проверку на пропуски `isnull()` выполним дальше, отсюда можно убрать. Также рекомендуем выводить разные запросы в отдельных ячейках

Столбцы в `bo_data` содержат следующую информацию:

 - ID посетителя
 - название ресторана
 - день бронирования
 - время прихода 
 - планируемое количество посетителей

<h3><font color="FFB800">Комментарий ассистента</font></h3>

Через запятую здесь не очень воспринимается, лучше перечислить данные через буллиты:

* ...
* ...

### Выводы

* Для работы с датасетом потребуется подготовка и преообразование данных, есть поле времени, текстовые значения дней недели и наименований ресторанов. ID посетителя не потребуется, кроме как для исключения дубликатов.

<h3><font color="89DD00">Комментарий ассистента</font></h3>

Все верно, в выводах обращаем внимание на структуру данных, дубликаты и пропуски. Единственное, попробуй все-таки метод `info()`.

## Предобработка данных

Получим названия столбцов.

In [5]:
bo_data.columns

Index(['userID ', ' ресторан ', ' weekday', ' ordered ', ' Столик для '], dtype='object')

Обращаем внимание, что помимо названий на разном языке, в них есть пробелы, которые затрудняют доступ к данным. Переименуем столбцы и проверим результат.

In [6]:
bo_data.set_axis(['user_ID', 'rest_name', 'weekday', 'ordered_time', 'visitor_num'], axis='columns',  inplace = True)
bo_data

user_ID   rest_name    weekday ordered_time  visitor_num
0     b912c3c6a6ad   Iskra Bar     Monday        10:00            1
1     2dac0150717d   Iskra Bar     Monday        13:30            2
2     ba83aefab5dc   Iskra Bar    Tuesday        12:30            3
3     02996f1dd2ea   Iskra Bar  Wednesday        12:00            2
4     7c90be56d2ab   Iskra Bar   Thursday        14:30            3
...            ...         ...        ...          ...          ...
1019  4b136de2c16a   Iskra Bar   Saturday        11:30            2
1020  9e46b91cc547  Бар Сибирь   Thursday        10:00            2
1021  5ca79042c44d      Там Ем   Thursday        14:00            3
1022  e0f9dfd1726c      Там Ем    Tuesday        17:30            1
1023  401d5fa8e55d      Там Ем  Wednesday        20:30            1

[1024 rows x 5 columns]

<h3><font color="#FFB800">Комментарий ассистента</font></h3>

Не надо отдельно удалять пробелы. Все столбцы можно переименовать с помощью одного метода — `set_axis()`. Передавай ему список с названиями столбцов, `axis='columns'`, `inplace = True` (последний аргумент — переименовывает именно существующий датафрейм, а не создает копию). Если хочешь переименовать только некоторые, то `rename()` подойдет, да, но опять же указывай `inplace = True`.

userID — это camel case, а rest_name — snake_case. Лучше на чем-то одном остановиться и еще time назвать очевиднее: например, ordered_time.

In [7]:
bo_data.columns

Index(['user_ID', 'rest_name', 'weekday', 'ordered_time', 'visitor_num'], dtype='object')

Еще раз проверим данные на наличие пропусков, просуммировав их.

In [8]:
bo_data.isnull().sum()

user_ID          0
rest_name       65
weekday          0
ordered_time     0
visitor_num      0
dtype: int64

Менеджер проекта предупреждал, что некоторые заказы заводились на тестовую карточку кафе без названия, поэтому есть смысл заполнить такой пропуск словом admin.

In [9]:
bo_data['rest_name']=bo_data['rest_name'].fillna(value="admin")

Проверяем, что пропусков не осталось.

In [10]:
bo_data.isnull().sum()

user_ID         0
rest_name       0
weekday         0
ordered_time    0
visitor_num     0
dtype: int64

Установим количество явных дубликатов. Если найдутся, то удалим и снова проверим, что их не осталось.

In [11]:
bo_data.duplicated().sum()

24

In [12]:
bo_data=bo_data.drop_duplicates().reset_index(drop=True)

In [13]:
bo_data.loc[20:27]

user_ID   rest_name    weekday ordered_time  visitor_num
20  ea12aec32ec3     Северия    Tuesday        12:30            4
21  838a82cec0a6       admin  Wednesday        13:00            4
22  de310e6aae86     Северия   Thursday        12:00            2
23  08b24e170109     Северия   Saturday        11:00            3
24  065a4daba35f  Бар Сибирь    Tuesday        13:30            1
25  990113ae4f1c  Крафтейнер     Monday        13:30            1
26  78164bc09657      Там Ем     Sunday        15:30            1
27  fbe29970a8a5  Бар Сибирь     Friday        16:00            3

<h3><font color="#FF0F00">Комментарий ассистента</font></h3>

Вывел срез, в котором были дубликаты, чтобы показать, что если не использовать метод `reset_index()`, то получается дыра. Этому методу также нужно передать `drop=True`, чтобы он создавал не отдельный столбец, а обновлял именно текущие индексы. Пиши цепочкой прямо `drop_duplicates().reset_index(drop=True)`.

<h3><font color="#000FFF">Комментарий студента</font></h3>

Исправил, здесь кажется не критично, т.к. обращаться по индексу не собираемся, но на будущее конечно следует сбрасывать.

In [14]:
bo_data.duplicated().sum()

0

Скорее всего, такие полные дубликаты говорят о сбое при выгрузке данных. Необходимо сообщить об этом.

Еще одна проблема, о которой предупреждал менеджер проекта — дубликаты карточки Iskra Bar с неправильными названием бара - @@. На них заводились реальные заказы, поэтому необходимо вычислить их и переименовать на правильное название — Iskra Bar.

Объявим функцию `replace_restaurant()`, которая заменяет неправильное название, а потом проверяет успешность выполнения замены подсчетом (если замена прошла успешно — результат подсчета должен быть 0).

In [15]:
'''
def replace_restaurant(name):
    df.loc[df['rest_name'] == name, 'rest_name'] = 'Iskra Bar'
    print(sum(df['rest_name'].isin({name})))
    return df
'''
def replace_restaurant1(df, wrong):
    df['rest_name'] = df['rest_name'].replace(wrong, 'Iskra Bar')
    result = df[df['rest_name'] == wrong]['rest_name'].count()
    print(result)
    return df

<h3><font color="#FFB800">Комментарий ассистента</font></h3>

Посмотри вот такую реализацию:

```
def replace_restaurant(df, wrong):
    df['restaurant'] = df['restaurant'].replace(wrong, 'Iskra Bar')
    result = df[df['restaurant'] == wrong]['restaurant'].count()
    
    return result
```

Все можно решить с помощью метода `replace()`, а потом "проверить" его выполнение с помощью `count()`. Рекомендую отдавать предпочтение именно методам pandas, а `sum()` или `len()` использовать меньше.

Также разделяй ячейки по логике выполнения команд — уникальные рестораны до выполнения функции функции лучше отдельно проверить.


<h3><font color="#000FFF">Комментарий студента</font></h3>

Не был знаком с `replace()`, учел.
Возвращать result некорректно, иначе потеряем весь прогресс.

Передаем функции названия 'Искра' и 'Искр'.

In [16]:
bo_data=replace_restaurant1(bo_data, 'Искра')
bo_data=replace_restaurant1(bo_data, 'Искр')

0
0


In [17]:
bo_data['rest_name'].unique()

array(['Iskra Bar', 'Там Ем', 'admin', 'Крафтейнер', 'Бар Сибирь',
       'Северия', 'Коворкафе', 'Азия'], dtype=object)

### Выводы

* Произведена предобработка данных: наименования столбцов приведены к общему стилю, удалены задвоенные значения, некорректные наименования одного из баров приведены к общему наименованию, тестовое заведение обозначено admin.

<h3><font color="#FFB800">Комментарий ассистента</font></h3>

Выводы к шагу хорошие, поправь только все, о чем я выше написал.

## Распределение заказов в течение недели

Считаем, на какие кафе приходится больше всего заказов.

In [18]:
bo_data['rest_name'].value_counts()

Iskra Bar     287
Там Ем        200
Бар Сибирь    159
Северия       107
Крафтейнер     78
Коворкафе      66
Азия           62
admin          41
Name: rest_name, dtype: int64

Группируем по `weekday` и считаем, на какие дни приходится больше всего заказов.

In [43]:
bo_data.groupby('weekday').size().sort_values(ascending=False)

weekday
Tuesday      251
Wednesday    242
Thursday     180
Monday       141
Friday       110
Saturday      52
Sunday        24
dtype: int64

<h3><font color="#89DD00">Комментарий ассистента</font></h3>

Да, со днями недели тоже можно разобраться с помощью `value_counts()` на самом деле, просто хотелось побольше группировки добавить. Поправим описание шага тут

<h3><font color="#000FFF">Комментарий студента</font></h3>

Переформулировал, суть не изменилась.

Объявляем функцию, которая возвращает количество заказов для ресторана в определенный день.

In [20]:
def orders_by_day(day, rest):
    orders=bo_data[(bo_data['rest_name'] ==rest) & (bo_data['weekday'] ==day)]
    orders_count = orders['rest_name'].count()
    print(f"В {rest} на {day} заказали  {orders_count} столик(ов)!")
    return (orders_count)
orders_by_day("Monday", "Iskra Bar")

В Iskra Bar на Monday заказали  48 столик(ов)!


48

<h3><font color="#FFB800">Комментарий ассистента</font></h3>

Хорошо, что логическое условие используешь, только я рекомендую снять `len()` и использовать `count()`:

```
def orders_by_day(day, rest):
    orders = df[(df['rest_name'] == rest) & (df['weekday'] == day)]
    orders_count = orders['rest_name'].count()
```

Передадим ей заведения Iskra Bar и Там Ем — два флагманских места, по которым просили результаты прежде всего — и понедельник, среду и пятницу в качестве дня недели.

In [21]:
m_is=orders_by_day("Monday", "Iskra Bar")

В Iskra Bar на Monday заказали  48 столик(ов)!


In [22]:
w_is=orders_by_day("Wednesday", "Iskra Bar")

В Iskra Bar на Wednesday заказали  53 столик(ов)!


In [23]:
f_is=orders_by_day("Friday", "Iskra Bar")

В Iskra Bar на Friday заказали  29 столик(ов)!


In [24]:
m_tam=orders_by_day("Monday", "Там Ем")

В Там Ем на Monday заказали  25 столик(ов)!


In [25]:
w_tam=orders_by_day("Wednesday","Там Ем")

В Там Ем на Wednesday заказали  52 столик(ов)!


In [26]:
f_tam=orders_by_day("Friday", "Там Ем")

В Там Ем на Friday заказали  24 столик(ов)!


Сведем результат в таблицу **(в которой столбцами являются ресторан и 3 дня недели!)**. 

In [27]:
'''
tableI = pd.DataFrame([["Iskra Bar", m_is, w_is , f_is]],
                   columns=["rest_name",'Monday','Wednesday',"Friday"])
tableT=pd.DataFrame([["Там Ем", m_tam, w_tam , f_tam]],
                   columns=["rest_name",'Monday','Wednesday',"Friday"])
tableI = pd.concat([tableI,tableT], axis=0, ignore_index=1)
tableI
'''
data = [
    ['Iskra Bar', m_is, w_is, f_is],
    ['Там Ем', m_tam, w_tam, f_tam]
]

columns = ['rest_name', 'Monday', 'Wednesday', 'Friday']

table = pd.DataFrame(data=data, columns=columns)
print(table)

   rest_name  Monday  Wednesday  Friday
0  Iskra Bar      48         53      29
1     Там Ем      25         52      24


<h3><font color="#FFB800">Комментарий ассистента</font></h3>

Результат правильный, но сведение в таблицу усложнил, посмотри вот такой подход:

```
data = [
    ['Iskra Bar', m_is, w_is, f_is],
    ['Там Ем', m_tam, w_tam, f_tam]
]

columns = ['rest_name', 'Monday', 'Wednesday', 'Friday']

result = pd.DataFrame(data=data, columns=columns)
print(table)
```

<h3><font color="#000FFF">Комментарий студента</font></h3>
Кажется не корректным использовать переменые data=data, columns=columns,
Как ни странно, составить таблицу оказалось одним из затруднительных моментов.

### Выводы

* Больше всего заказов по приложению: Искра и Там ем. Вне контекста от потока/наличия посадочных мест/популярности места.
* Суммарно чаще всего предзаказывают столики во вторник и среду, во флагманских местах также преобладает среда, но отдельно стоит Iskra Bar - понедельник практически на уровне среды.


<h3><font color="#FFB800">Комментарий ассистента</font></h3>

«Наиболее посещаемые» — спорная категория. Азия может иметь 60 посадочных мест и заполнять их на 100%, а Искра 500 и только половину. Если сравнивать так, то Азия более посещаемая


<h3><font color="#000FFF">Комментарий студента</font></h3>

Исправил, без расширенных данных возможны только прямые ответы и меньше размышлений.

## Распределение заказов в течение дня

Сравним количества заказов, сделанных в обеденное и вечернее время — с 12 до 16 и с 16 до 22. Для этого объявим функцию, которая принимает данные и 2 аргумента времени.

In [28]:

'''    df['time'] =  pd.to_datetime(df['ordered_time'], format='%H:%M')
    time1=dt.strptime(time1, '%H:%M')
    time2=dt.strptime(time2, '%H:%M')
    orders = sum(df['ordered_time'].between(time1, time2, inclusive=True))
    print(f"C {time1.hour} до {time2.hour} заказали {orders} столик(ов)!")
    return 
'''
def orders_by_day(df, time1, time2):
    ord_=df[(df['ordered_time'] >= time1) & (df['ordered_time'] < time2)]
    ord_day=ord_['ordered_time'].count()
    print(ord_day)
orders_by_day(bo_data, "12:00", "16:00")

543


Передадим промежуток с 12:00 и 16:00

In [29]:
orders_by_day(bo_data, "12:00", "16:00")

543


И с 16:00 до 22:00

In [30]:
orders_by_day(bo_data, "16:00", "22:00")

338


<h3><font color="FFB800">Комментарий ассистента</font></h3>

Круто, что пришел к мысли преобразовать время! В этом проекте мы не закладывали такой подход, поэтому я рекомендую пока использовать логическое условие как и раньше `df[(df['column1'] == value1) & (df['column2'] == value2)]` и в конце прицеплять к нему метод `count()`. Если сравнивать сами строки без преобразования — результат тоже правильный будет.


<h3><font color="#000FFF">Комментарий студента</font></h3>

Внезапно всё проще, исправил так же учет времени до 16:00.

### Выводы

* Разница между "бизнес ланчами" и вечерними предзаказми практически в 2 раза в пользу обеденного времени.

<h3><font color="#89DD00">Комментарий ассистента</font></h3>

Вывод здесь главный в том, что большинство заказов делают именно в обед. Это то, что и просили от нас узнать на текущем этапе, все верно.

## Типичное количество человек за столиком

Выясним, для какого количества человек обычно заказывают столик в разное время на примере Iskra Bar и Там Ем. Для этого напишем функцию, которая принимает данные, название ресторана, 2 времени и считает моду.

In [31]:
'''
def moda_orders(df, rest, time1, time2):
    df['time'] =  pd.to_datetime(df['time'], format='%H:%M')
    time1=dt.strptime(time1, '%H:%M')
    time2=dt.strptime(time2, '%H:%M')
    orders = (df[(df['rest_name'] ==rest) & (df['time'].between(time1, time2, inclusive=True))])
    mode_ = statistics.mode(orders['Visitor_Num'])
    print(f"Чаще всего в {rest} с {time1.hour} до {time2.hour} заказывали столики на {mode_} ({(orders['Visitor_Num'] == mode_).sum()} раз)!") 
moda_orders(df, 'Iskra Bar', "12:00", "16:00")
'''

'\ndef moda_orders(df, rest, time1, time2):\n    df[\'time\'] =  pd.to_datetime(df[\'time\'], format=\'%H:%M\')\n    time1=dt.strptime(time1, \'%H:%M\')\n    time2=dt.strptime(time2, \'%H:%M\')\n    orders = (df[(df[\'rest_name\'] ==rest) & (df[\'time\'].between(time1, time2, inclusive=True))])\n    mode_ = statistics.mode(orders[\'Visitor_Num\'])\n    print(f"Чаще всего в {rest} с {time1.hour} до {time2.hour} заказывали столики на {mode_} ({(orders[\'Visitor_Num\'] == mode_).sum()} раз)!") \nmoda_orders(df, \'Iskra Bar\', "12:00", "16:00")\n'

<h3><font color="FFB800">Комментарий ассистента</font></h3>

Максим, здесь в точности такой же подход может быть — логическое условие и пандосовский `mode()` на конце. Модуль `statistics` не обязателен (кстати, импорт всех модулей стоит производить в самом начале ноутбука (программного кода).

И еще смотрю, ты ставишь `inclusive=True`, но по идее заказ в 16:00 уже к относится к вечернему времени.

Попробуй прямо в ячейке ниже написать `moda_orders_2()` с учетом моих комментов.

In [32]:
def moda_orders_2(df, rest, time1, time2):
    ord_in_time=df[(df['ordered_time'] >= time1) & (df['ordered_time'] < time2)]
    orders = ord_in_time[(ord_in_time['rest_name'] ==rest) ]
    #print(orders)   
    mod =orders['visitor_num'].mode()
    print(mod[0])   
    print(f"Чаще всего в {rest} с {time1} до {time2} заказывали столики на {mod[0]} ({(orders['visitor_num'] == mod[0]).sum()} раз)!") 
moda_orders_2(bo_data, 'Iskra Bar', "12:00", "16:00") 

2
Чаще всего в Iskra Bar с 12:00 до 16:00 заказывали столики на 2 (72 раз)!


Проверим функцию на двух ресторанах, а также дневном и вечернем промежутке времени.

In [33]:
moda_orders_2(bo_data, 'Iskra Bar', "12:00", "16:00")

2
Чаще всего в Iskra Bar с 12:00 до 16:00 заказывали столики на 2 (72 раз)!


In [34]:
moda_orders_2(bo_data, 'Там Ем', "12:00", "16:00")

2
Чаще всего в Там Ем с 12:00 до 16:00 заказывали столики на 2 (43 раз)!


In [35]:
moda_orders_2(bo_data, 'Iskra Bar', "16:00", "22:00")

1
Чаще всего в Iskra Bar с 16:00 до 22:00 заказывали столики на 1 (45 раз)!


In [36]:
moda_orders_2(bo_data, 'Там Ем', "16:00", "22:00")

1
Чаще всего в Там Ем с 16:00 до 22:00 заказывали столики на 1 (34 раз)!


### Выводы

* Самые популярные к заказу столики в дневное время - на 2х, вечером на 1, на основании чего можно сделать вывод о наиболее частой брони столиков для обедов и ужинов, чем на большую компанию. 

<h3><font color="#89DD00">Комментарий ассистента</font></h3>

Все именно так. As simple as that!

## Общие выводы

#### На какие кафе приходится больше всего заказов? А на какие дни?
Наибольшее число заказов приходится на Iskra Bar - 287 и Там Ем - 200.
Основные посещения приходятся на рабочую неделю, но самые посещаемые дни в середине недели вторник - 251, среда - 242, четверг - 180. По общей статистике понедельник и пятницу можно считать среднезагруженными, а выходные низкозагруженными. Флагманские места могут иметь более повышенные показатели всю рабочую неделю. 

#### Чаще ли приложением пользуются именно в обеденное время?
Предзаказ в обеденное время преобладает для сохранения времени, суммарно за неделю практически в 2 раза даже с учетом выходных.

#### Насколько большими компаниями (или наоборот поодиночке) приходят в кафе?
На примере популярных мест: чаще всего бронируют столики в обеденное время на двоих, в вечернее на одного.



Приложение полезно ограниченным по времени посетителям (1-2 человека), предугадать же запросы на большую компанию заказчику может быть сложно. 

<h3><font color="#89DD00">Комментарий ассистента</font></h3>

«Наибольшее число заказов приходится» — а вот здесь уже корректная формулировка, все четко.

### Комментарий по проекту

Из-за различий описания задач в брифе и notebook задания могут быть двусмысленны (шаг 4 -переделывал таблицу, шаг 6 так и не понял верно сделал или нет). Бриф подразумевается как предварительный текст задания или основной?

Датасет по итогу включает всех посетителей, даже не пользующихся приложением?

<h3><font color="#89DD00">Комментарий ассистента</font></h3>

В брифе — задача от школы, а в ноутбуке идут «описания» действий, которые мы помогли вам сформулировать. 

Для построения ds-портфолио важны 2 вещи — блог, то есть умение рассказать о своем "ходе мышления" на проекте и код в репозиториях на GitHub. Формат ноутбука является 2 в 1 и сейчас вы (с нашей помощью) описываете в нем свой проект именно для потенциального работодателя.

По брифу уже видими слабые места, доработаем.

По вопросу про датасает, в описании задачи мы пишем: «Необходимо изучить данные сервиса...», так что, конечно, только те, кто пользуется приложением.
